### PR TITLE
Removed deprecated features and notices

### PR DESCRIPTION
**Pull request by javiereguiluz**
@xabbuh and @weaverryan to avoid a nightmare or rebases and problems, could we please give top priority to merging this PR? Thanks! It doesn't have to be perfect or 100% completed. We can do more fixes in another PR. Thanks!

-----

Some articles explain things that are removed. Instead of removing those articles entirely and to avoid user confusion, I propose to keep those articles but explain that those features no longer exist. For example, `security/acl.rst` now contains the following:

```rst
.. index::
   single: Security; Access Control Lists (ACLs)

How to Use Access Control Lists (ACLs)
======================================

.. caution::

    ACL support was removed in Symfony 4.0. Install the `Symfony ACL bundle`_
    and refer to its documentation if you want to keep using ACL.

.. _`Symfony ACL bundle`: https://github.com/symfony/acl-bundle
```

Articles like this one:

* security/acl.rst
* security/acl_advanced.rst
* components/event_dispatcher/container_aware_dispatcher.rst

We can remove all these deprecated articles in 4.1

-----

@iltar could you please create a PR to update the changes related to "controller arguments" in these two articles? Thanks!

* `components/http_kernel.rst`
* `create_framework/http_kernel_controller_resolver.rst`

I don't understand the changes well (`ControllerResolverInterface` removed `getArguments()`)

-----

@HeahDude could you please create a PR to update the changes needed about the deprecation of using `false` in `property_path` instead of `mapped` in `reference/forms/types/options/property_path.rst.inc`? Thanks!

-----

There is a pending deprecation message in `security/multiple_user_providers.rst` ... but that's fixed in the pending PR #8612.

-----

A final question: I've removed the explanation of the deprecated `kernel.root_dir` in `reference/configuration/kernel.rst` and kept only `kernel.project_dir`.  Would you instead prefer to keep a reference to the still heavily used `kernel.root_dir` to explain that it's deprecated?

### CODE REVIEW

**Comment by iltar** [comment]
I'll take a look on Monday

File: bundles/extension.rst [hunk 1]
```diff
@@ -129,17 +129,10 @@ read more about it, see the ":doc:`/bundles/configuration`" article.
 Adding Classes to Compile
 -------------------------
 
-.. versionadded:: 3.3
-    This technique is discouraged and the ``addClassesToCompile()`` method was
-    deprecated in Symfony 3.3 because modern PHP versions make it unnecessary.
```
Comment (wouterj) [inline bundles/extension.rst]: Unless I'm wrong, this versionadded says "Don't use what's descriped below." (so it's more a `.. caution::` than versionadded). This means the article should be removed (as the feature got removed in Symfony 4).

File: bundles/override.rst [hunk 2]
```diff
@@ -5,8 +5,8 @@ How to Override any Part of a Bundle
 ====================================
 
 This document is a quick reference for how to override different parts of
-third-party bundles without using :doc:`/bundles/inheritance`, which is
-deprecated since Symfony 3.4.
+third-party bundles without using :doc:`/bundles/inheritance`, which was
+removed in Symfony 4.0.
```
Comment (wouterj) [inline bundles/override.rst]: Let's make this description generic to not require any more updates in later versions (and thus merge conflicts), e.g:

 > When using another bundle, you might want to customize or override some of its features. This document describes ways of overriding the most common features of a bundle.

File: components/console/events.rst [hunk 3]
```diff
@@ -83,27 +83,24 @@ C/C++ standard.::
...
+thrown by the application.
 
 Listeners receive a
 :class:`Symfony\\Component\\Console\\Event\\ConsoleExceptionEvent` event::
 
     use Symfony\Component\Console\Event\ConsoleExceptionEvent;
     use Symfony\Component\Console\ConsoleEvents;
 
-    $dispatcher->addListener(ConsoleEvents::EXCEPTION, function (ConsoleExceptionEvent $event) {
+    $dispatcher->addListener(ConsoleEvents::ERROR, function (ConsoleExceptionEvent $event) {
```
Comment (wouterj) [inline components/console/events.rst]: Typehint should be `ConsoleErrorEvent`

File: components/console/events.rst [hunk 4]
```diff
@@ -117,19 +114,6 @@ Listeners receive a
         $event->setException(new \LogicException('Caught exception', $exitCode, $event->getException()));
```
Comment (wouterj) [inline components/console/events.rst]: `$event->getError()`

File: components/event_dispatcher/container_aware_dispatcher.rst [hunk 5]
```diff
@@ -4,103 +4,7 @@
...
-            // ...
-        }
-
-        public function onStoreOrder(FilterOrderEvent $event)
-        {
-            // ...
-        }
-    }
+    The ``ContainerAwareEventDispatcher`` was removed in Symfony 4.0. Use
+    ``EventDispatcher`` with closure-proxy injection instead.
```
Comment (wouterj) [inline components/event_dispatcher/container_aware_dispatcher.rst]: reference article/section about closure-proxy injection?

File: profiler/matchers.rst [hunk 6]
```diff
@@ -6,158 +6,5 @@ How to Use Matchers to Enable the Profiler Conditionally
...
-        $container->loadFromExtension('framework', array(
-            // ...
-            'profiler' => array(
-                'matcher' => array(
-                    'service' => SuperAdminMatcher::class,
-                )
-            ),
-        ));
+    The possibility to use a matcher to enable the profiler conditionally was
+    removed in Symfony 4.0.
```
Comment (wouterj) [inline profiler/matchers.rst]: We should add information about doing `$profiler->disable()` in your controller explicitly if you need this functionality.

**Review by wouterj — APPROVED** [review]
Left comments, but can be done after merging.

File: bundles/extension.rst (hunk 1 above)
Comment (javiereguiluz) [reply bundles/extension.rst]: I though the same ... but only `addClassesToCompile()` is deprecated/removed. `addAnnotatedClassesToCompile()` is still valid and used in Sf4.

File: components/event_dispatcher/container_aware_dispatcher.rst (hunk 5 above)
Comment (javiereguiluz) [reply components/event_dispatcher/container_aware_dispatcher.rst]: I don't understand this comment. Is there an article that we could link to in this comment? Or are you suggesting to create a new article? Thanks!

**Comment by weaverryan** [comment]
+1 for removing `kernel.root_dir` entirely. It will be easy for people to find it on the internet anyways.

**Comment by weaverryan** [comment]
I've just merged this so we can keep conflicts low and keep moving quickly (thanks Wouter for the fast review). I did get the tests passing.

If anyone *does* see any issues, please open a separate PR! 

@javiereguiluz can you open separate issues on the 4.0 milestone for the other things that you needed done by other people?

Thanks!

**Comment by javiereguiluz** [comment]
Done! Thanks for your great and quick reviews and for merging this. Cheers!

**Pull request merged**